### PR TITLE
TS 4.8.0+ Add implementation of new methods for lates typescript

### PR DESCRIPTION
**Pull request by elkeis**
The problem: 

If you update typescript, or use `sta` in your fresh project:

```sh
npm update typescript > /dev/null && cat package-lock.json | grep typescript
```

```sh                                                                                              
  "name": "swagger-typescript-api",
      "name": "swagger-typescript-api",
        "typescript": "^4.8.3"
        "swagger-typescript-api": "index.js"
    "node_modules/typescript": {
      "resolved": "https://registry.npmjs.org/typescript/-/typescript-4.8.3.tgz",
    "typescript": {
      "resolved": "https://registry.npmjs.org/typescript/-/typescript-4.8.3.tgz",
```

You got an error if you are trying to put some extras into your `generated`.

```sh
npm run test:--axios          
```

``` sh                                                                                                                                           
> swagger-typescript-api@10.0.1 test:--axios
> node tests/spec/axios/test.js

/Users/elkeis/work/github.com/elkeis:sta/src/index.js:267
          throw new Error("Swagger schema parse error!\r\n " + e);
                ^

Error: Swagger schema parse error!
 TypeError: host.fileExists is not a function
    at /Users/elkeis/work/github.com/elkeis:sta/src/index.js:267:17
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
```

This PR fixes this issue by adding new methods to LanguageServiceHost implementation for Formatters.

### CODE REVIEW

File: src/formatFileContent.js [hunk 1]
```diff
@@ -39,7 +39,21 @@ class LanguageServiceHost {
   getScriptSnapshot() {
     return ts.ScriptSnapshot.fromString(this.content);
   }
-}
+  readFile(fileName, encoding) {
+      if (fileName === this.fileName) {
+        return this.content
+      } else {
+        return ts.sys.readFile(fileName, encoding);
+      }
```
Comment (csi-lk) [inline src/formatFileContent.js:47]: nitpick; unnecessary else

```suggestion
      if (fileName === this.fileName) {
        return this.content
      }
      
      return ts.sys.readFile(fileName, encoding);
```

File: src/formatFileContent.js [hunk 2]
```diff
@@ -39,7 +39,21 @@ class LanguageServiceHost {
...
+  }
+  fileExists(fileName) {
+      if (fileName === this.fileName) {
+        return true;
+      } else {
+        return ts.sys.fileExists(fileName);
+      }
```
Comment (csi-lk) [inline src/formatFileContent.js:54]: nitpick; unnecessary else

```suggestion
      if (fileName === this.fileName) {
        return true;
      }
      
      return ts.sys.fileExists(fileName);
```

**Review by csi-lk — APPROVED** [review]
Looks good to me, straightforward update, just needs to be merged

**Comment by danielantelo** [comment]
@csi-lk any updates on when this will be released?

**Comment by jfstn** [comment]
I came across this issue and I wasn't able to use yarn's module resolution properly in a monorepo.

You can save yourself some time if you are stuck until we have a new release by changing the corresponding line in your `package.json` to the one below 👇 

```
    "swagger-typescript-api": "git+https://github.com/elkeis/swagger-typescript-api.git#bug/latest-typescript-fileExists-issue",
```

**Pull request closed without merging**